### DISCUSSION TITLE
CCIP-012 / CCIP-013 Remaining Work Items

### DISCUSSION THREAD

**Discussion by whoabuddy** [citycoins/governance#9] [CCIPs]
# CCIP-012 / CCIP-013 Remaining Work Items

---

*Note: for a more recent list of tasks see citycoins/protocol/issues/34*

---

This discussion summarizes the remaining tasks for each phase of CCIP-012/013, and will serve as the main place to see status updates throughout the voting and implementation.

- [CCIP-012: Stabilize Emissions and Treasuries](https://github.com/citycoins/governance/blob/main/ccips/ccip-012/ccip-012-stabilize-emissions-and-treasuries.md)
- [CCIP-013: Stabilize Protocol and Simplify Contracts](https://github.com/citycoins/governance/blob/feat/stabilize-protocol/ccips/ccip-013/ccip-013-stabilize-protocol-and-simplify-contracts.md)

## Preparing for the Vote

Check that all repositories are merged and up-to-date.

[citycoins/contracts](https://github.com/citycoins/contracts)
- [x] citycoins/contracts#204
- [x] citycoins/contracts#205
- [x] tag main with new [version 2.0.0](https://github.com/citycoins/contracts/tree/v2.0.0)
- [x] make `develop` default branch

[citycoins/ui](https://github.com/citycoins/ui)
- [x] citycoins/ui#156
- [x] make `develop` default branch

[citycoins/api](https://github.com/citycoins/api)
- [x] good to go!

[citycoins/governance](https://github.com/citycoins/governance)
- [x] add CCIP authors and supporting materials
- [x] finalize CCIP-012 details
- [x] update README with activation details
- [x] check/fix citycoins/governance#7

---

Add Ryan Waits (StackerDAO Labs) to the GitHub org.

- [x] done!

Create voting contract for CCIP-012

- [x] citycoins/contracts#207 *(finalized, awaiting review)*
    - [x] blocker: citycoins/contracts#208
    - [x] workaround: API can query historical info
- [x] determine vote scale factor ([V1 and V2 scale factor calculations](https://docs.google.com/spreadsheets/d/1F1RGDXF2AO7J0yGGfxAYPrn2lBPT9yZ0LjpDgEkcPmc/edit#gid=0))
  - MIA cycles 21 and 22
  - NYC cycles 15 and 16
  - MIA scale factor: 0.8605 (prev: 0.6987)
- [x] test with new tardis contract ([deployed](https://explorer.stacks.co/txid/SP2NS7CNBBN3S9J6M4JJHT7WNBETRSBZ9KPVRENBJ.citycoin-tardis-v3?chain=mainnet))
- [x] include references to CCIP-012 *(finalized version)*
- [x] fix read-only contract call budget issues
- [x] simplify and update contract code
- [x] update CCIP-011 with vote details and spreadsheet
- [x] create timeline graphic (see Miro for v1 example)

Create voting UI to interact with contract

- [x] get access to repository with previous UI framework
- [x] decide domain to deploy - stabilize.minecitycoins.com ?
- [x] update vote UI information to new contract
- [x] deploy new vote UI at stabilize.minecitycoins.com

## Phase 1: Reduce CityCoin Emissions

- [x] finalize 2% emissions modeling ([linked here](https://docs.google.com/spreadsheets/d/17aF6LlniJ3Bk3YomDynOHRrx4vxiyotplD42hUbiSRA/edit?usp=sharing))
- [x] convert spreadsheet to ods format, store with CCIP-012
- [x] add table to CCIP with block rewards v1-v2-v3
- [x] simulate upcoming changes citycoins/contracts#209
- [x] finalize [Miro outline of implementation flow](https://miro.com/app/board/o9J_lGTdbbg=/?moveToWidget=3458764530753789579&cot=14)

For MIA/NYC implementation:

- [x] create job in auth contract
- [x] add uint parameters for coinbase amounts
- [x] [compare amounts to existing results](https://github.com/citycoins/scripts/blob/feat/ccip-012/src/ccip-012-job-check.ts)
- [x] activate job for approval voting
- [x] get 3/5 approval
- [x] execute job

## Phase 2: Move CityCoin Treasuries to Smart Contract Vaults

- [x] create new [citycoins/protocol](https://github.com/citycoins/protocol) repo to store DAO contracts

Plan initial DAO structure with 3-of-5 signers

- [x] develop launch plan similar in structure to the [EcosystemDAO launch plan](https://www.notion.so/claritylab/EcosytemDAO-Launch-plan-9309928b675e429993f9b1b108119694)
- [x] collab with StackerDAOs, EcosystemDAO

Implement initial DAO proposals

- [x] citycoins/protocol#1
  - [x] base-dao.clar
  - [x] ccd001-direct-execute.clar
  - [x] ccd002-treasury-mia.clar
  - [x] ccd002-treasury-nyc.clar
  - [x] double-check all contract interactions
- [x] citycoins/protocol#2 (in final review)
  - [x] add relevant contract tests
  - [x] request peer reviews on final design
  - [x] generate deployer address: SP1XQXW9JNQ1W4A7PYTN3HCHPEY7SHM6KP98H3NCY
- [x] implement fixes for tests https://github.com/citycoins/protocol/pull/9
- [x] deploy DAO contracts to testnet

Update the wallet address in core contracts

- [x] uses 3-of-5 auth job like the coinbase amounts
- [x] simulate upcoming changes citycoins/contracts#209
- [x] review list of impacted entities [similar to this one](https://citycoins.wiki/citycoins-emissions-upgrade)
  - CityCoins UI (minecitycoins.com)
  - Syvita UI (syvitamining.com)
  - miamining/mining.nyc dashboards
  - Twitter bots or anything that reports wallet balances

## Phase 3: Simplify the CityCoins Contract Design

- [x] deploy and activate CCIP-013 vote https://github.com/citycoins/contracts/pull/210
- [x] develop simplified version to perform initial testing
  - [x] ccd003-user-registry.clar
  - [x] ccd004-city-registry.clar
  - [x] ccd005-citycoin-mining.clar
  - [x] ccd006-citycoin-stacking.clar
- [x] research better migration paths for mining/stacking data
  - mining should not miss any blocks due to code-as-proposal implementation
    - could activate only if previous contract is shut down first?
  - stacking should migrate state between contracts for the current cycle
    - or refer back to the original until the active cycle is complete
- [x] setup proposal with Clarity to execute that:
  - [x] sets up and enables the extensions
  - [x] initializes any contracts in the protocol
  - *note that both above could be separated if needed*
- [x] research old issues / improvements to incorporate
- [x] opportunity for code cleanup / flow control checks (simplify!)
- [x] convert from current auth patterns to Executor DAO extensions
  - [x] generalize as a "legacy connector"?
  - [x] coinbase amounts
  - [x] coinbase thresholds
  - [x] set token URI
  - [x] set city wallet

## Phase 4: Update Registration, Mining and Stacking Flows

- [x] develop design for current stacking model (pooled delegation)
- [x] prepare design for Stacks 2.1 launch (Nov-Dec 2022)

### Design Goals

- 100% of STX spent mining CityCoins is transferred to the city’s treasury
- 100% of the STX within the city’s treasury are stacked for STX rewards
- 100% of the STX rewards are claimable by anyone who stacks CityCoins, such that:
  - rewards are distributed per past stacking cycles
  - rewards are distributed proportionally to the amount of CityCoins stacked
  - if they are not claimed within n cycles, they can be claimed by the DAO? (open for discussion)

### Reward Cycle Participation - MIA

Reward Cycle | Total Stacked MIA | Total Supply MIA | Percentage
-- | -- | -- | --
16 | 3,337,657,374 | 4,825,550,000 | 69.17%
17 | 903,513,362 | 4,997,150,000 | 18.08%
18 | 1,101,162,929 | 5,100,150,000 | 21.59%
19 | 1,751,845,435 | 5,204,100,000 | 33.66%
20 | 2,274,467,272 | 5,310,300,000 | 42.83%
21 | 2,469,371,221 | 5,411,200,000 | 45.63%
22 | 2,593,145,882 | 5,516,000,000 | 47.01%

### Reward Cycle Participation - NYC

Reward Cycle | Total Stacked NYC | Total Supply NYC | Percentage
-- | -- | -- | --
10 | 2,803,440,578 | 3,558,250,000 | 78.79%
11 | 1,128,337,110 | 3,747,150,000 | 30.11%
12 | 1,514,271,882 | 3,961,550,000 | 38.22%
13 | 2,371,806,059 | 4,164,750,000 | 56.95%
14 | 2,625,652,238 | 4,384,250,000 | 59.89%
15 | 2,980,119,157 | 4,598,450,000 | 64.81%
16 | 3,036,819,828 | 4,805,950,000 | 63.19%

## Nice to Haves

[citycoins/scripts](https://github.com/citycoins/scripts)

- [x] remove references to documentation
- [x] remove workflow for documentation build
- [x] release V2 scripts from `feat/implement-v2`

**Comment by Rapha-btc** [comment]
Hello Jason, 

do you know when Stacks will have moved to delegated / smart contract and when it will be time to do stacking with new contract? 

Thank you.
